regionmask/defined_regions/from_geopandas.py

```diff
@@ -0,0 +1,126 @@
+import numpy as np
+import pandas as pd
+
+from ..core.regions import Regions
+from .natural_earth import _maybe_get_column
+
+
+def _check_duplicates(to_check):
+    """Checks if `to_check` has duplicates. If so, raises error. If not, return True."""
+    if len(set(to_check)) == len(to_check):
+        return True
+    else:
+        if isinstance(to_check, list):
+            duplicates = set([x for x in to_check if to_check.count(x) > 1])
+        elif isinstance(to_check, pd.core.series.Series):
+            duplicates = to_check[to_check.duplicated(keep=False)]
+        else:
+            raise ValueError(
+                "to_check not in [list, pd.Series], found {}".format(type(to_check))
+            )
+        raise ValueError("Found duplicates {}. {}".format(duplicates, to_check))
+
+
+def _construct_abbrevs(geodataframe, names):
+    """Construct unique abbreviations based on geodataframe.names."""
+    if names is None:
+        raise ValueError(
+            "names is None, but should be a valid column name of"
+            "geodataframe, choose from {}".format(geodataframe.columns)
+        )
+    abbrevs = []
+    for item in geodataframe.T:
+        name_for_abbrev = geodataframe.loc[item][names]
```

Comment:
```suggestion
    names = geodataframe[names]
    names = names.str.replace("[ ()\.]", "")
    for name in names:
```
move the getitem out of the loop and leverage pandas for the string replacement